docs/testplan/MACsec-test-plan.md

```diff
@@ -0,0 +1,445 @@
+| policy                |                                                           *security*                                                            |
+| enable_replay_protect |                                                         *true*/*false*                                                          |
+| replay_window         |                                    enable_replay_protect(*0*)/disable_replay_protect(*100*)                                     |
+| send_sci              |                                                         *true*/*false*                                                          |
+| rekey_period          |                                                            *0*/*60*                                                             |
+
+***Port table***
+| Field               |                Value                |
+| ------------------- | :---------------------------------: |
+| pfc_encryption_mode | *bypass*/*encrypt*/*strict_encrypt* |
+
+**NOTE: The macsec functionality testcases will cover the various options for each of the above supported fields**
+
+## Test cases
+
+### Testcase : Macsec Functionality
+
+This testcase covers the macsec/MKA protocol functionality
+
+#### Check Control plane
+
+- Enable macsec on the interface on the DUT and remote VM.
+- Check the process, `wpa_supplicant`, for the target port is running in the devices.
+- Check APP DB, Check the following fields in MACsec port table are consistent with configuration
+
+    | Config DB Field | Config DB Value |  App DB Field  | APP DB Value |
+    | :-------------: | :-------------: | :------------: | :----------: |
+    |                 |                 |     enable     |     true     |
+    |  cipher_suite   |   GCM-AES-128   |  cipher_suite  | GCM-AES-128  |
+    |  cipher_suite   |   GCM-AES-256   |  cipher_suite  | GCM-AES-256  |
+    |                 |                 | enable_protect |     true     |
+    |     policy      |    security     | enable_encrypt |     true     |
+    |     policy      | integrity_only  | enable_encrypt |    false     |
+    |    send_sci     |      true       |    send_sci    |     true     |
+    |    send_sci     |      false      |    send_sci    |    false     |
+
+- Check the following fields in MACsec SC table and MACsec SA table are consistent
+
+    1. There should be a MACsec SA in MACsec SA table with the same AN of *encoding_an* in MACsec SC.
+    2. The count of ingress MACsec SA shouldn't be lesser than the count of egress MACsec SA in the peer side.
+    3. The corresponding ingress and egress MACsec SA should have same *sak* and *auth_key*.
+    4. The *next_pn* of egress MACsec SA shouldn't be lesser than the *lowest_acceptable_pn* of the corresponding ingress MACsec SA in the peer side.
+
+- Check MKA session
+
+    1. Get the MKA session by `ip macsec show`
```

Comment:
ip macsec show works only for Linux based macsec. It would be good to mention macsec show SONiC CLI for hardware based macsec.

Reply:
Thanks for this suggestion, Rephrased it.